src/python/zquantum/qaoa/recursive_qaoa.py

```diff
@@ -21,6 +22,8 @@ def __init__(
             [IsingOperator, Ansatz], EstimationTasksFactory
         ],
         cost_function_factory: Callable[[EstimationTasksFactory], CostFunction],
+        # TODO: is there a way to specify that cost_function_factory must have a keyword argument `estimation_tasks_factory`?
```

Comment:
I don't think it's possible in typing.
But you can add an assertion in the body of the function, sth like:
```
assert has_attr(cost_function_factory, estimation_tasks_factory)
```

Reply:
Or actually an Exception with more explicit message would be better.

Reply:
I think the way to do this is to use a [callback protocol](https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols).

Reply:
correct me if I'm wrong but it looks like `hasattr` doesn't work with functions, only instances of a class?

```
>>> hasattr(self._cost_function_factory, "estimation_tasks_factory") 
False
>>> from zquantum.core.cost_function import create_cost_function
>>> hasattr(create_cost_function, "estimation_tasks_factory")       
False